shadows/framework/src/main/java/org/robolectric/shadows/ShadowBiometricManager.java

```diff
@@ -65,9 +95,49 @@ public void setCanAuthenticate(boolean flag) {
     biometricServiceConnected = flag;
   }
 
+  /**
+   * @param num to set the authenticate result
+   */
+  public void setEnrolledBiometrics(int num) {
+    enrolledBiometrics = num;
+  }
+
+  /**
+   * @param flag to set has hardware or not
+   */
+  public void hasHardware(boolean flag) {
+    hasHardware = flag;
+  }
+
+  /**
+   * @param type to set the authenticatorType
+   */
+  public void setAuthenticatorType(int type) {
+    authenticatorType = type;
+  }
+
   @Implementation(minSdk = R)
   protected int canAuthenticate(int userId, int authenticators) {
-    return biometricServiceConnected ? BIOMETRIC_SUCCESS : BIOMETRIC_ERROR_NO_HARDWARE;
+    int biometricError = getBiometricError();
+    if (biometricError != NO_BIOMETRIC_ERROR) return biometricError;
+    if (biometricServiceConnected) return BIOMETRIC_SUCCESS;
+    return BIOMETRIC_ERROR_NO_HARDWARE;
+  }
+
+  private int getBiometricError() {
+    if (!biometricServiceConnected) return BIOMETRIC_ERROR_NO_HARDWARE;
```

Comment:
It's recommended to add {} for if statement, although it has only one line to keep consistent style of if statement.

Reply:
Sure no problem 🆗